src/v/archival/ntp_archiver_service.h

```diff
@@ -198,7 +198,7 @@ class ntp_archiver {
     ss::lowres_clock::duration _cloud_storage_initial_backoff;
     ss::lowres_clock::duration _segment_upload_timeout;
     ss::lowres_clock::duration _manifest_upload_timeout;
-    ss::semaphore _mutex{1};
+    ss::named_semaphore _mutex = ssx::make_named_semaphore(1, "ntp-archiver");
```

Comment:
I know it's not your code, but whilst you're here, should this be an `ssx::named_mutex`?

Reply:
+1 I think this would be a good followup PR.